src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -79,97 +89,140 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
-
-            var elementAccessExpressionsAndIndexerMemberCref = elementAccessExpressions.Concat(indexerMemberCref);
-
-            var locations = ArrayBuilder<ReferenceLocation>.GetInstance();
+            var locations = ArrayBuilder<FinderLocation>.GetInstance();
 
-            foreach (var node in elementAccessExpressionsAndIndexerMemberCref)
+            foreach (var node in elementAccessExpressions)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                var match = symbolsMatch(node, semanticModel);
-                if (match.matched)
+                var (matched, reason) = symbolsMatch(node, semanticModel);
+                if (matched)
                 {
-                    SyntaxNode nodeToBeReferenced = null;
+                    syntaxFacts.GetPartsOfElementAccessExpression(node, out var expression, out var argumentList);
 
-                    if (syntaxFacts.IsIndexerMemberCRef(node))
+                    if (symbolsMatch(expression, semanticModel).matched)
                     {
-                        nodeToBeReferenced = node;
-                    }
-                    else
-                    {
-                        var childNodes = node.ChildNodes();
-                        var leftOfInvocation = childNodes.First();
-                        if (symbolsMatch(leftOfInvocation, semanticModel).matched)
-                        {
-                            // Element access with explicit member name (allowed in VB).
-                            // We have already added a reference location for the member name identifier, so skip this one.
-                            continue;
-                        }
-
-                        nodeToBeReferenced = childNodes.Skip(1).FirstOrDefault();
+                        // Element access with explicit member name (allowed in VB).
+                        // We have already added a reference location for the member name identifier, so skip this one.
+                        continue;
                     }
 
-                    if (nodeToBeReferenced != null)
-                    {
-                        var location = nodeToBeReferenced.SyntaxTree.GetLocation(new TextSpan(nodeToBeReferenced.SpanStart, 0));
-                        locations.Add(new ReferenceLocation(document, null, location, isImplicit: false, isWrittenTo: false, candidateReason: match.reason));
-                    }
+                    var location = argumentList.SyntaxTree.GetLocation(new TextSpan(argumentList.SpanStart, 0));
+                    locations.Add(new FinderLocation(
+                        node, new ReferenceLocation(document, null, location, isImplicit: false, isWrittenTo: false, candidateReason: reason)));
+                }
+            }
+
+            return locations.ToImmutableAndFree();
+        }
+
+        private async Task<ImmutableArray<FinderLocation>> FindIndexerCrefReferencesAsync(
```

Comment:
💭 Why are indexer `cref` references different from other property `cref` references?

Reply:
i'm not sure i 100% understand your question.  However, effectively, the original code tried to share this codepath, but had to go do funky stuff in the middle.  It was pretty ugly and hard to reason about (Esp. when trying to figure out what would happen when the property-finder code was called from teh accessor-finder code).  So i broke it up into two simpler methods that were much easier to reason about.